### PR TITLE
issue #133 : add connectivity map option connectivityByDeployments

### PR DESCRIPTION
**Pull request by adisos**
- I will add some tests after the merge from issue #131, to avoid conflicts.
- I have added the outputConfig option "connectivityByDeployments". Should we add some new cli option, or is it sufficient to control through scheme files?

### CODE REVIEW

File: network-config-analyzer/ConnectivityGraph.py [hunk 1]
```diff
@@ -52,29 +65,24 @@ def get_connectivity_dot_format_str(self):
...
+            peer_name, is_ip_block = self._get_peer_name(peer)
+            peer_color = "red2" if is_ip_block else "blue"
+            peer_lines.append(f'\t\"{peer_name}\" [label=\"{peer_name}\" color=\"{peer_color}\" fontcolor=\"{peer_color}\"]\n')
 
-        edge_lines = []
+        edge_lines = set()
         for connections, peer_pairs in self.connections_to_peers.items():
             for src_peer, dst_peer in peer_pairs:
-                if src_peer != dst_peer and not connections.allow_all:
+                if src_peer != dst_peer and not connections.allow_all and connections:
```
Comment (adisos) [inline network-config-analyzer/ConnectivityGraph.py:75]: I think we should avoid adding edges with no connections.
Do we want to keep not adding edges with 'allow all'?

Comment (zivnevo) [reply network-config-analyzer/ConnectivityGraph.py:75]: Yes, we should avoid adding edges with no connections.
We do want to keep edges with 'allow all'. Why did we skip them anyway?

Comment (adisos) [reply network-config-analyzer/ConnectivityGraph.py:75]: I'm not sure why they were skipped.. I will change it now.

File: network-config-analyzer/ConnectivityGraph.py [hunk 2]
```diff
@@ -52,29 +65,24 @@ def get_connectivity_dot_format_str(self):
             output_result += f'\tHEADER [shape="box" label=< <B>{self.output_config.queryName}/{self.output_config.configName}</B> > fontsize=30 color=webmaroon fontcolor=webmaroon];\n'
         peer_lines = []
```
Comment (zivnevo) [inline network-config-analyzer/ConnectivityGraph.py:66]: Don't we want a set here too?

**Review by zivnevo — CHANGES_REQUESTED** [review]
ToDos:
- [ ] A cli flag to turn on connectivity-by-deployments
- [ ] Tests (waiting for updates to testing scheme)

File: network-config-analyzer/ConnectivityGraph.py (hunk 2 above)
Comment (adisos) [reply network-config-analyzer/ConnectivityGraph.py]: right -- fixed.

**Pull request merged**